crates/egui/src/containers/panel.rs

```diff
@@ -513,281 +518,91 @@ impl SidePanel {
-        self.height_range = height_range;
-        self
-    }
-
-    /// Enforce this exact height, including margins.
-    #[inline]
-    pub fn exact_height(mut self, height: f32) -> Self {
-        self.default_height = Some(height);
-        self.height_range = Rangef::point(height);
-        self
-    }
-
-    /// Change the background color, margins, etc.
-    #[inline]
-    pub fn frame(mut self, frame: Frame) -> Self {
-        self.frame = Some(frame);
-        self
-    }
-}
-
-impl TopBottomPanel {
-    /// Show the panel inside a [`Ui`].
-    pub fn show_inside<R>(
-        self,
-        ui: &mut Ui,
-        add_contents: impl FnOnce(&mut Ui) -> R,
-    ) -> InnerResponse<R> {
-        self.show_inside_dyn(ui, Box::new(add_contents))
-    }
-
+// Private methods to support the various show methods
+impl Panel {
     /// Show the panel inside a [`Ui`].
     fn show_inside_dyn<'c, R>(
         self,
         ui: &mut Ui,
         add_contents: Box<dyn FnOnce(&mut Ui) -> R + 'c>,
     ) -> InnerResponse<R> {
-        let Self {
-            side,
-            id,
-            frame,
-            resizable,
-            show_separator_line,
-            default_height,
-            height_range,
-        } = self;
-
-        let frame = frame.unwrap_or_else(|| Frame::side_top_panel(ui.style()));
-
-        let available_rect = ui.available_rect_before_wrap();
-        let mut panel_rect = available_rect;
-
-        let mut height = if let Some(state) = PanelState::load(ui.ctx(), id) {
-            state.rect.height()
-        } else {
-            default_height
-                .unwrap_or_else(|| ui.style().spacing.interact_size.y + frame.inner_margin.sum().y)
+        let side = self.side;
+        let id = self.id;
+        let resizable = self.resizable;
+        let show_separator_line = self.show_separator_line;
+        let size_range = self.size_range;
+
+        // Define the sizing of the panel.
+        let mut panel_sizer = PanelSizer::new(&self, ui);
+
+        // Check for duplicate id
+        ui.ctx()
+            .check_for_id_clash(id, panel_sizer.panel_rect, "Panel");
+
+        if self.resizable {
+            // Prepare the resizable panel to avoid frame latency in the resize
+            self.prepare_resizable_panel(&mut panel_sizer, ui);
+        }
+
+        // NOTE(shark98): This must be **after** the resizable preparation, as the size
+        // may change and round_ui() uses the size.
+        panel_sizer.panel_rect = panel_sizer.panel_rect.round_ui();
+
+        let get_ui_kind = || match side {
+            PanelSide::Vertical(v_side) => match v_side {
+                VerticalSide::Left => UiKind::LeftPanel,
+                VerticalSide::Right => UiKind::RightPanel,
+            },
+            PanelSide::Horizontal(h_side) => match h_side {
+                HorizontalSide::Top => UiKind::TopPanel,
+                HorizontalSide::Bottom => UiKind::BottomPanel,
+            },
         };
```

Comment:
Is there a reason this is a closure?

```suggestion
        let ui_kind = match side {
            PanelSide::Vertical(v_side) => match v_side {
                VerticalSide::Left => UiKind::LeftPanel,
                VerticalSide::Right => UiKind::RightPanel,
            },
            PanelSide::Horizontal(h_side) => match h_side {
                HorizontalSide::Top => UiKind::TopPanel,
                HorizontalSide::Bottom => UiKind::BottomPanel,
            },
        };
```

Reply:
Not that I can think of. Like above, being totaly new to Rust, I may have overthinked some stuff here and there (although I see that I did put the two indent of match together and not as an external closure 😅, I guess I was being more confortable once I got here!).

Also, I did use RustRover refactoring to extract as a variable in many places and sometime it used a closure, sometime not. But I am not knowledgeable enough in Rust to provide concrete and technical reasoning being those choices.

I'll look into it with all the others.